src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -212,40 +212,58 @@ private Version getServerVersion() throws IOException {
     NodesInfo info = new NodesInfo.Builder().addCleanApiParameter("version").build();
     JsonObject result = client.execute(info).getJsonObject();
     if (result == null) {
-      LOG.warn("Couldn't get Elasticsearch version, result is null");
+      LOG.warn("Couldn't get Elasticsearch version (result is null); assuming {}", defaultVersion);
       return defaultVersion;
     }
 
     checkForError(result);
 
     JsonObject nodesRoot = result.get("nodes").getAsJsonObject();
     if (nodesRoot == null || nodesRoot.entrySet().size() == 0) {
-      LOG.warn("Couldn't get Elasticsearch version, nodesRoot is null or empty");
+      LOG.warn(
+          "Couldn't get Elasticsearch version (response nodesRoot is null or empty); assuming {}",
+          defaultVersion
+      );
       return defaultVersion;
     }
 
     JsonObject nodeRoot = nodesRoot.entrySet().iterator().next().getValue().getAsJsonObject();
     if (nodeRoot == null) {
-      LOG.warn("Couldn't get Elasticsearch version, nodeRoot is null");
+      LOG.warn(
+          "Couldn't get Elasticsearch version (response nodeRoot is null); assuming {}",
+          defaultVersion
+      );
       return defaultVersion;
     }
 
     String esVersion = nodeRoot.get("version").getAsString();
+    Version version;
     if (esVersion == null) {
-      LOG.warn("Couldn't get Elasticsearch version, version is null");
-      return defaultVersion;
+      version = defaultVersion;
+      LOG.warn(
+          "Couldn't get Elasticsearch version (response version is null); assuming {}",
+          version
+      );
     } else if (esVersion.startsWith("1.")) {
-      return Version.ES_V1;
+      version = Version.ES_V1;
+      log.info("Detected Elasticsearch version is {}", version);
     } else if (esVersion.startsWith("2.")) {
-      return Version.ES_V2;
+      version = Version.ES_V2;
+      log.info("Detected Elasticsearch version is {}", version);
     } else if (esVersion.startsWith("5.")) {
-      return Version.ES_V5;
+      version = Version.ES_V5;
+      log.info("Detected Elasticsearch version is {}", version);
     } else if (esVersion.startsWith("6.")) {
-      return Version.ES_V6;
+      version = Version.ES_V6;
+      log.info("Detected Elasticsearch version is {}", version);
     } else if (esVersion.startsWith("7.")) {
-      return Version.ES_V7;
+      version = Version.ES_V7;
+      log.info("Detected Elasticsearch version is {}", version);
+    } else {
+      version = defaultVersion;
+      log.info("Detected unexpected Elasticsearch version {}, using {}", esVersion, version);
     }
-    return defaultVersion;
+    return version;
```

Comment:
I think one log message can subsume the rest of these
```suggestion
    log.info("Detected Elasticsearch version is {}", version);
    return version;
```

Reply:
Only if we don't want to know about the final `else` block case. I was trying not to restructure this code to make the logging simpler, and preferred to leave the logic untouched and to have extra logging code.

Reply:
The logic has already been refactored in this PR, you've changed the `return` flow. I agree that it's the same control flow overall, and I don't think it should be put back the way it was. I just don't think that it's a valid argument as to why you wouldn't refactor the log messages.

You can leave the else log message, since that's valuable. I would even make it a WARN level.

Reply:
The WARN log messages were there before; I just clarified the message enough that they now need to span multiple lines. And it's a warning because the behavior is a function of the version, so if we can't get an actual version then the behavior may not be correct. It's not an error because we don't know for sure.

I've tried restructuring this section of code, and I just don't see an easy way to do it that keeps the code readable and straightforward.